pymtl3/passes/VcdGenerationPass.py

```diff
@@ -81,6 +79,57 @@ def _gen_vcd_symbol():
         yield code
         n += 1
 
+    # The actual dump_vcd function to be appended to _sched.
+    # The framework will call this function without arguments, and therefore
+    # I chose to monkey patch the data structure objects to the global
+    # namespace of `dump_vcd`. This might be hacky but probably better than
+    # compiling a million lines of code to a Python function (you will almost
+    # certainly get a segfault from that).
+    # TODO: make this less hacky? type check?
+
+    def dump_vcd():
+      s             = top
+      vcd_file      = vcdmeta.vcd_file
+      clock_net_idx = vcdmeta.clock_net_idx
+      clock_symbol  = net_symbol_mapping[vcdmeta.clock_net_idx]
+      next_neg_edge = 100*vcdmeta.sim_ncycles+50
+      next_pos_edge = 100*vcdmeta.sim_ncycles+100
+
+      try:
```

Comment:
As far as I can see, you are turning the dynamic compilation for each signal into a loop that iterates over all the provided signals. I think the loop approach is better than code generation, since the "simulation performance improvement" that dynamic compilation is supposed to provide is unclear here -- we might want to see if dumping vcd is bottlenecked by disk (which I think so).